src/modules/visits/services/visits.service.ts

```diff
@@ -274,44 +275,55 @@ export class VisitsService {
 			}
 		}
 
-		if (updateVisit.access_type) {
-			// IF accessFolderIds is empty and accessType is FOLDERS
-			// OR accessFolderIds is not empty and accessType is FULL
-			// THEN throw BadRequest exception
-			if (
-				isEmpty(accessType) ||
-				(isEmpty(accessFolderIds) && updateVisit.access_type === VisitAccessType.Folders) ||
-				(!isEmpty(accessFolderIds) && updateVisit.access_type === VisitAccessType.Full)
-			) {
-				throw new BadRequestException(
-					`The amount of accessFolderIds (${accessFolderIds.length}) does not correspond with the accessType ${updateVisit.access_type}`
-				);
+		//IF status is DENIED
+		//THEN change access_type to FULL and clear accessFolderIds
+		if (updateVisit.status === VisitStatus.DENIED) {
+			updateVisit.access_type = VisitAccessType.Full;
+			accessFolderIds = null;
+		} else {
+			if (updateVisit.access_type) {
```

Comment:
Wouldn't it be easier to do the following instead of change the user input?

```
if (updateVisit.access_type) {
	// Always delete the old access folders
	await this.dataService.execute<
		DeleteVisitFolderAccessMutation,
		DeleteVisitFolderAccessMutationVariables
	>(DeleteVisitFolderAccessDocument, {
		visitRequestId: currentVisit.id,
	});

	if (updateVisit.status !== VisitStatus.DENIED) {
		// IF accessFolderIds is empty and accessType is FOLDERS
		// OR accessFolderIds is not empty and accessType is FULL
		// THEN throw BadRequest exception
		if (
			isEmpty(accessType) ||
			(isEmpty(accessFolderIds) &&
			updateVisit.access_type === VisitAccessType.Folders) ||
			(!isEmpty(accessFolderIds) && updateVisit.access_type === VisitAccessType.Full)
		) {
			throw new BadRequestException(
				`The amount of accessFolderIds does not correspond with the accessType ${updateVisit.access_type}`
			);
		}

	// IF accessFolderIds is not empty and accessType is FOLDERS
	// THEN add new folders to the folder_access table
	if (
		!isEmpty(accessFolderIds) &&
		updateVisit.access_type === VisitAccessType.Folders
	) {
		await this.dataService.execute<
			InsertVisitFolderAccessMutation,
			InsertVisitFolderAccessMutationVariables
		>(InsertVisitFolderAccessDocument, {
			objects: [
				...accessFolderIds.map((accessFolderId: string) => ({
					folder_id: accessFolderId,
					visit_request_id: currentVisit.id,
				})),
			],
		});
	}
    }
}
```

@bertyhell thoughts? I just don't like setting the value after the user input

Reply:
i told Emile to do it that way, because i wanted to avoid folder ids in the database that are linked to denied visit requests.
Since that could maybe cause issues on the elasticsearch filtering side, if we don't correctly filter visit requests.

So if a visit is denied, i would always set the type to full and delete the folders.

@tourlbr Do you see any advantage in keeping those folder ids around?

Reply:
@bertyhell No, I agree those def need to be removed